grammar.js

```diff
@@ -2485,6 +2459,15 @@ function unsigned_type($, type) {
   )
 }
 
+function optional_parenthesis(node) {
+  return prec.right(
```

Comment:
does it not work without right-precedence?

Reply:
Nope, I think it's because there's multiple layers where the parenthesis are allowed to appear so this pushes it "down" the tree.

```
$ tree-sitter generate
Unresolved conflict for symbol sequence:

  '('  union  •  ')'  …

Possible interpretations:

  1:  '('  (_dml_read  union)  •  ')'  …
  2:  (_dml_read  '('  union  •  ')')

Possible resolutions:

  1:  Specify a left or right associativity in `_dml_read`
  2:  Add a conflict for these rules: `_dml_read`
```

Reply:
ah weird, that's probably it. Not a lot to do about it I suppose